include/reffine/engine/cudaengine.h

```diff
@@ -0,0 +1,38 @@
+#ifndef INCLUDE_REFFINE_CUDA_ENGINE_H_
+#define INCLUDE_REFFINE_CUDA_ENGINE_H_
+
+#include "llvm/ExecutionEngine/Orc/JITTargetMachineBuilder.h"
+#include "llvm/IR/DataLayout.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/IR/Module.h"
+#include "llvm/MC/TargetRegistry.h"
+#include "llvm/Support/TargetSelect.h"
+
+using namespace std;
+using namespace llvm;
+
+namespace reffine {
+
+class CUDAEngine {
```

Comment:
first read my comment on the ExecutePTX implementation.

Reply:
After you have moved ExecutePTX() out of this, I think we should make this yet another code generation pass, similar to LLVMGen.

Reply:
For now, to keep things simple, I would suggest to add another LLVMGen::Build() function to generate and return ptx as a string.

Reply:
then we don't need CUDAEngine at all.